core/src/main/java/org/apache/iceberg/view/BaseSQLViewRepresentation.java

```diff
@@ -0,0 +1,206 @@
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder buildFrom(SQLViewRepresentation representation) {
+    return builder()
+        .query(representation.query())
+        .schemaId(representation.schemaId())
+        .dialect(representation.dialect())
+        .defaultCatalog(representation.defaultCatalog())
+        .defaultNamespace(representation.defaultNamespace())
+        .fieldAliases(representation.fieldAliases())
+        .fieldComments(representation.fieldComments());
+  }
+
+  private BaseSQLViewRepresentation(
+      String query,
+      String dialect,
+      String defaultCatalog,
+      Namespace defaultNamespace,
+      int schemaId,
+      List<String> fieldAliases,
+      List<String> fieldComments) {
+    this.query = Preconditions.checkNotNull(query, "sql should not be null");
+    this.dialect = Preconditions.checkNotNull(dialect, "dialect should not be null");
+    this.defaultCatalog =
+        Preconditions.checkNotNull(defaultCatalog, "default catalog should not null");
+    this.defaultNamespace =
+        Preconditions.checkNotNull(defaultNamespace, "default namespace should not be null");
+    this.schemaId = Preconditions.checkNotNull(schemaId, "schema should not be null");
+    this.fieldAliases =
+        Preconditions.checkNotNull(fieldAliases, "field aliases should not be null");
+    this.fieldComments =
+        Preconditions.checkNotNull(fieldComments, "field comments should not be null");
+  }
+
+  @Override
+  public String query() {
+    return query;
+  }
+
+  @Override
+  public String dialect() {
+    return dialect;
+  }
+
+  @Override
+  public String defaultCatalog() {
+    return defaultCatalog;
+  }
+
+  @Override
+  public Namespace defaultNamespace() {
+    return defaultNamespace;
+  }
+
+  @Override
+  public int schemaId() {
+    return schemaId;
+  }
+
+  @Override
+  public List<String> fieldComments() {
+    return fieldComments;
+  }
+
+  @Override
+  public List<String> fieldAliases() {
+    return fieldAliases;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    BaseSQLViewRepresentation that = (BaseSQLViewRepresentation) o;
+    return Objects.equals(query, that.query)
+        && Objects.equals(dialect, that.dialect)
+        && Objects.equals(schemaId, that.schemaId)
+        && Objects.equals(defaultCatalog, that.defaultCatalog)
+        && Objects.equals(defaultNamespace, that.defaultNamespace)
+        && Objects.equals(fieldAliases, that.fieldAliases)
+        && Objects.equals(fieldComments, that.fieldComments);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        query, dialect, schemaId, defaultCatalog, defaultNamespace, fieldAliases, fieldComments);
+  }
+
+  @Override
+  public String toString() {
+    return "BaseViewDefinition{"
```

Comment:
Can you use a helper rather than building this by hand? It's hard to read this way. We usually base these on `MoreObjects`.